### PR TITLE
bootstrapping Javascript build step

### PR DESCRIPTION
**Pull request by davidmsibley**
Lots of things going on in this one, I'll walk you through the noteworthy parts:

### Add package.json
Lets start with the build step. I added a `package.json` set up similar to other projects, with some additions. The normal development process still applies: run an `npm start` to start up a live development server.

However, because the javascript needs to be transpiled before it can run, the normal "github pages from master" scheme needed to change. I've added a script to build and push a gh-pages branch, you can run it via `npm run pages`. I intend to set this up on Jenkins to run after every commit.

### Introducing Rollup
I've brought in and configured a bundling tool called [Rollup](https://rollupjs.org/). It will go though the source code, gather all the javascript files, transpile them, bundle them all into one file, and create a minified version as well. In prior projects I've simply just run a minifier on the one or two javascript files that were there, but if we want to structure our code into more than a couple files, it'll be easier to just let Rollup handle it.

### Transpiling?
It means taking one version of javascript and changing it into another. In this case, taking fancy ES Module code like in [modules/app-sidebar.js][] and changing it into a version of Javascript that the majority of browsers will understand. I'm using [Babel](https://babeljs.io/) for this.

### Including HTML Fragments (via Web Components)
Alongside `app-sidebar.js` there's a [modules/app-sidebar.html][]. This HTML Fragment is included into the main page via a custom element `<app-sidebar></app-sidebar>`. Inside the fragment, I'm binding [Mustache](http://mustache.github.io/) templates to the state within `app-sidebar.js`. I've added a title attribute, a couple of dynamic fields and a slot to just demo the capabilities.  I envision using this style of module creation to build out the data elements and the filter interface.

This is built on top of a small base library that I put together yesterday: [AppComponent](https://github.com/davidmsibley/app-component).  
AppComponent draws from prior work that I did with others at DoIT in the pursuit of easy framework-less [Web Components](https://en.wikipedia.org/wiki/Web_Components). I included it here on a whim. I really like how it turned out, but there could be better long term options:

To get an industry standard version of similar functionality, I'd recommend [Vue JS](https://vuejs.org/). ... I say "similar," I mean: Vue JS has this and waaaay more. But, you're also tying yourself to a framework, so there's tradeoffs.

If we only want to include static HTML Fragments (don't need the Mustache binding), then we may want to try out a GitHub produced web component: [include-fragment-element](https://github.com/github/include-fragment-element). I just recently stumbled upon a stash of GitHub web components and they look pretty solid, I've been itching to try them out.

### In conclusion...
Those are the main parts of this PR. We can chat about this in more detail whenever you'd like.

[modules/app-sidebar.js]: https://github.com/davidmsibley/springs-data/blob/e38a019364f7a3e8dcedae418aab6f7a14120d66/modules/app-sidebar.js
[modules/app-sidebar.html]: https://github.com/davidmsibley/springs-data/blob/e38a019364f7a3e8dcedae418aab6f7a14120d66/modules/app-sidebar.html

### CODE REVIEW

**Comment by davidmsibley** [comment]
You can run through the demo here: https://davidmsibley.github.io/springs-data/
